### PR TITLE
Docs: Doxygen Documentation Deployment

### PR DESCRIPTION
**Pull request by jinzx10**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### What's changed?
A new workflow that deploys the doxygen-generated html documentation as github pages is added. The workflow will be triggered whenever there is a push event (like the merge of a PR) to the "develop" branch.

Update:
The workflow is configured such that only one deployment can exist at any time, i.e., if multiple PRs are merged within a short period of time, only the one triggered by the last merge will be executed; those triggered by previous PRS will be cancelled.

A preview of the documentation can be found at https://jinzx10.github.io/abacus-develop/

### CODE REVIEW

**Review by caic99 — APPROVED** [review]

File: doxygen/Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2582 @@
...
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = ABACUS
+
+# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         = 2024-05-29
```
Comment (caic99) [inline doxygen/Doxyfile]:
```suggestion
PROJECT_NUMBER         = develop
```
This appears at the head line, which may confuses the user.

**Review by mohanchen — APPROVED** [review]
Nice feature.

File: doxygen/Doxyfile (hunk 1 above)
Comment (jinzx10) [reply doxygen/Doxyfile]: advice followed!

**Comment by jinzx10** [comment]
The workflow is configured such that only one can exist at any time. If multiple PRs are merged within a short period of time, while each of them will trigger the deployment action, only the one triggered by the last merge will be fully executed; those by the previous PRs will be automatically cancelled, even in progress.

**Pull request merged**